File: subprojects/docs/src/samples/userguide/plugins/propertyServiceInjection/java/buildSrc/src/main/java/UrlProcess.java
import javax.inject.Inject;
import org.gradle.api.model.ObjectFactory;
import org.gradle.api.DefaultTask ;
import org.gradle.api.tasks.TaskAction;
import org.gradle.workers.WorkerExecutor;

public abstract class UrlProcess extends DefaultTask {
    // Use an abstract getter method
    @Inject
    protected abstract ObjectFactory getObjectFactory();

    // Alternatively, use a getter method with a dummy implementation
    @Inject
    protected WorkerExecutor getWorkerExecutor() {
        // Method body is ignored
        throw new UnsupportedOperationException();
    }

    @TaskAction
    void run() {
        WorkerExecutor workerExecutor = getWorkerExecutor();
        ObjectFactory objectFactory = getObjectFactory();
        // Use the executor and factory ...
    }
}

File: subprojects/docs/src/samples/userguide/plugins/propertyServiceInjection/kotlin/buildSrc/src/main/kotlin/UrlProcess.kt
import javax.inject.Inject
import org.gradle.api.model.ObjectFactory
import org.gradle.api.DefaultTask
import org.gradle.api.tasks.TaskAction
import org.gradle.workers.WorkerExecutor

abstract class UrlProcess : DefaultTask() {
    // Use an abstract property
    // Note that the @Inject annotation must be attached to the getter
    @get:Inject
    abstract val objectFactory: ObjectFactory

    // Alternatively, use a property getter with a dummy implementation
    // Note that the property must be open and the @Inject annotation must be attached to the getter
    @get:Inject
    open val workerExecutor: WorkerExecutor
        get() {
            // Getter body is ignored
            throw UnsupportedOperationException()
        }

    @TaskAction
    fun run() {
        // Use the executor and factory ...
    }
}
